### PR TITLE
Added unit tests for low-coverage System.Windows.Forms classes

### LINKED ISSUE #13442: Ensure that Button-related classes under the namespace "System.Windows.Forms" have at least 80% unit test coverage
Refer to https://app.codecov.io/gh/dotnet/winforms/tree/main/src%2FSystem.Windows.Forms%2FSystem%2FWindows%2FForms%2FControls%2FButtons
- [ ] ButtonBaseAdapter.ColorOptions.cs  
- [ ] ButtonBaseAdapter.LayoutOptions.cs  
- [ ] ButtonBaseAdapter.cs  
- [ ] ButtonFlatAdapter.cs  
- [ ] ButtonPopupAdapter.cs  
- [ ] ButtonStandardAdapter.cs  
- [ ] CheckBoxBaseAdapter.cs  
- [ ] CheckBoxFlatAdapter.cs  
- [ ] CheckBoxPopupAdapter.cs  
- [ ] CheckBoxStandardAdapter.cs  
- [ ] CheckableControlBaseAdapter.cs  
- [ ] RadioButtonBaseAdapter.cs  
- [ ] RadioButtonFlatAdapter.cs  
- [ ] RadioButtonPopupAdapter.cs  
- [ ] RadioButtonStandardAdapter.cs  
- [ ] ApplicableToButtonAttribute.cs  
- [ ] ButtonRenderer.cs  
- [ ] CheckBox.cs  
- [ ] CheckBoxRenderer.cs  
- [ ] FlatButtonAppearance.cs  
- [ ] FlatButtonAppearanceConverter.cs  
- [ ] RadioButtonRenderer.cs

### PR DESCRIPTION
**Pull request by Sathish-087**
Fixes [#13442](https://github.com/dotnet/winforms/issues/13442)
 
## Proposed changes
- Added 9 unit tests for RadioButtonFlatAdapter.
- Added 11 unit tests for RadioButtonPopupAdapter.
- Added 8 unit tests for ToolStrip.RestoreFocusMessageFilter.
- Added 8 unit tests for HtmlHistory.
- Covered paint operations (PaintUp, PaintDown, PaintOver) for radio button adapter implementations.
- Added coverage for adapter creation and layout behavior validation.
- Added tests for DPI-aware check size calculations and shadowed text behavior.
- Added coverage for checked-state rendering and text/image rendering scenarios.
- Added tests validating focus restoration and message filtering behavior in ToolStrip.RestoreFocusMessageFilter.
- Added tests covering HtmlHistory navigation APIs (Back, Forward, Go), property accessors, argument validation, and disposal behavior.
- Improved unit test coverage for low-coverage classes in the System.Windows.Forms namespace.
 
## Customer Impact
- Improves unit test coverage for several previously low-coverage System.Windows.Forms classes.
- Increases confidence in rendering, layout, focus restoration, and HTML history navigation functionality.
- Helps prevent regressions in public and protected virtual APIs across the tested components.
 
## Regression?
- No
 
## Risk
- Low (tests only, no product code changes)
 
## Test methodology
- Executed existing WinForms unit test suite.
- Added and executed new unit tests for RadioButtonFlatAdapter.
- Added and executed new unit tests for RadioButtonPopupAdapter.
- Added and executed new unit tests for ToolStrip.RestoreFocusMessageFilter.
- Added and executed new unit tests for HtmlHistory.
 
## Accessibility testing
NA
 
## Test environment(s)
- Visual Studio (Test Explorer)
- Windows
- .NET 11
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/14911)

### CODE REVIEW

**Comment by KlausLoeffelmann** [comment]
@SimonZhao888, could you do a manual review of this? Thx!

File: src/test/unit/System.Windows.Forms/System/Windows/Forms/HtmlHistoryTests.cs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        };
+
+        HtmlDocument document = await GetDocument(control, HtmlPage1);
+        document.Window.Should().NotBeNull();
+        using HtmlHistory? history = document.Window.History;
+
+        history.Should().NotBeNull();
+
+        // Relative position always forwards to COM go(); 0 is the safe relative position.
+        Action goRelative = () => history!.Go(0);
```
Comment (SimonZhao888) [inline src/test/unit/System.Windows.Forms/System/Windows/Forms/HtmlHistoryTests.cs:175]: It would make our test cases more comprehensive if we could also cover some edge-case navigation scenarios here—such as `Go(-1)` or cases where the index falls outside the range of available history.

File: src/test/unit/System.Windows.Forms/System/Windows/Forms/RadioButtonFlatAdapterTests.cs [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    public void PaintDown_DoesNotThrow(Appearance appearance, bool enabled, bool @checked)
+    {
+        (RadioButtonFlatAdapter adapter, RadioButton control) = CreateAdapter(appearance, enabled, @checked);
+        using Bitmap bitmap = new(control.Width, control.Height);
+        using Graphics graphics = Graphics.FromImage(bitmap);
+        using PaintEventArgs e = new(graphics, control.ClientRectangle);
+
+        Action action = () => adapter.PaintDown(e, control.Checked ? CheckState.Checked : CheckState.Unchecked);
+
+        action.Should().NotThrow();
```
[the remark this answers is not in the document]
Comment (Sathish-087) [reply src/test/unit/System.Windows.Forms/System/Windows/Forms/RadioButtonFlatAdapterTests.cs:72]: Updated the paint test cases based on existing "HasPixelChanged" pattern.

File: src/test/unit/System.Windows.Forms/System/Windows/Forms/RadioButtonPopupAdapterTests.cs [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    public void PaintDown_DoesNotThrow(Appearance appearance, bool enabled, bool @checked)
+    {
+        (RadioButtonPopupAdapter adapter, RadioButton control) = CreateAdapter(appearance, enabled, @checked);
+        using Bitmap bitmap = new(control.Width, control.Height);
+        using Graphics graphics = Graphics.FromImage(bitmap);
+        using PaintEventArgs e = new(graphics, control.ClientRectangle);
+
+        Action action = () => adapter.PaintDown(e, control.Checked ? CheckState.Checked : CheckState.Unchecked);
+
+        action.Should().NotThrow();
```
[the remark this answers is not in the document]
Comment (Sathish-087) [reply src/test/unit/System.Windows.Forms/System/Windows/Forms/RadioButtonPopupAdapterTests.cs:79]: Updated the paint test cases based on existing HasPixelChanged pattern.

File: src/test/unit/System.Windows.Forms/System/Windows/Forms/HtmlHistoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+    private static async Task NavigateToPathAsync(WebBrowser control, string path)
+    {
+        TaskCompletionSource<bool> source = new();
+        void Handler(object? sender, WebBrowserDocumentCompletedEventArgs e) => source.TrySetResult(true);
+        control.DocumentCompleted += Handler;
+        try
+        {
+            await Task.Run(() => control.Navigate(path));
+            Assert.True(await source.Task);
```
[the remark this answers is not in the document]
Comment (Sathish-087) [reply src/test/unit/System.Windows.Forms/System/Windows/Forms/HtmlHistoryTests.cs]: Modified this test case

File: src/test/unit/System.Windows.Forms/System/Windows/Forms/HtmlHistoryTests.cs (hunk 1 above)
Comment (Sathish-087) [reply src/test/unit/System.Windows.Forms/System/Windows/Forms/HtmlHistoryTests.cs:175]: Added new test case for Go(-1).
